WBooks/src/redux/search/reducer.js

```diff
@@ -0,0 +1,17 @@
+import Immutable from 'seamless-immutable';
+
+import { actionTypes } from './constants/actionTypes';
+
+const initialState = {
+  text: '',
+  books: []
+};
+
+export const searchReducer = (state = Immutable(initialState), action) => {
```

Comment:
Why don't you use redux recompose?

Reply:
It's a very simple action without much logic. And there is not failure, get or success actions